08-callyourapi/app/src/main/java/auth0/callyourapidemo/activities/MainActivity.java

```diff
@@ -0,0 +1,140 @@
+
+public class MainActivity extends AppCompatActivity {
+
+    private Button mAuthenticatedRequestButton;
+    private Button mNonAuthenticatedRequestButton;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        mAuthenticatedRequestButton = (Button) findViewById(R.id.authenticatedButton);
+        mNonAuthenticatedRequestButton = (Button) findViewById(R.id.nonAuthenticatedButton);
+
+
+        mAuthenticatedRequestButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                authenticateWithTokenID();
+            }
+        });
+
+        mAuthenticatedRequestButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                authenticateWithoutTokenID();
+            }
+        });
+
+
+
+
+    }
+
+    /**
+     * This method request should work fine, if your server configuration is ok
+     */
+
+    private void authenticateWithTokenID() {
+
+        // Instantiate the RequestQueue.
+        RequestQueue queue = Volley.newRequestQueue(this);
+        String url = "your api url"; // TODO Replace this
+
+        AuthorizationRequestObject authorizationRequest = new AuthorizationRequestObject
+                (Request.Method.GET, url, App.getInstance().getUserCredentials().getIdToken(),
+                        null, new Response.Listener<JSONObject>() {
+
+                    @Override
+                    public void onResponse(JSONObject response) {
+                        // Parse Response
+                    }
+                }, new Response.ErrorListener() {
+
+                    @Override
+                    public void onErrorResponse(VolleyError error) {
+
+                    }
+                });
+
+        // Add the request to the RequestQueue.
+        queue.add(authorizationRequest);
+
+    }
+
+
+    /**
+     * This method request should fail, if your server configuration is ok
+     */
+
+    private void authenticateWithoutTokenID() {
+
+        // Instantiate the RequestQueue.
+        RequestQueue queue = Volley.newRequestQueue(this);
+        String url = "your api url"; // TODO Replace this
+
+        AuthorizationRequestObject authorizationRequest = new AuthorizationRequestObject
+                (Request.Method.GET, url, "",
+                        null, new Response.Listener<JSONObject>() {
+
+                    @Override
+                    public void onResponse(JSONObject response) {
+                        // Parse Response
+                    }
+                }, new Response.ErrorListener() {
+
+                    @Override
+                    public void onErrorResponse(VolleyError error) {
+
+                    }
+                });
+
+        // Add the request to the RequestQueue.
+        queue.add(authorizationRequest);
+
```

Comment:
- Can you make it a single method that sends authentication or not depending on a parameter?

- The app should show an alert or something saying that the request failed or succeeded. Show the Status Code or something of the returned response.

- Shouldn't it be formated as "Bearer {Token}" in the "Authentication" header?